lib/game.rb

```diff
@@ -0,0 +1,59 @@
+require '../lib/interface.rb'
+
+class Game
+  include Interface
+  attr_accessor :symbol
+
+  def initialize(board, player_1, player_2)
+    @board = board
+    @player_1 = player_1
+    @player_2 = player_2
+    @turn = 1
+   end
+
+  def win?(item)
+    temp = []
+    @board.board.each_with_index do |value, index|
+      temp << index if value == item
+    end
+    @board.win_patterns.each do |arr|
+      result = arr - temp
+      return true if result.empty?
+    end
+    false
+  end
+
+  def add_at(pos, symbol)
+    @board.board[pos] = symbol
+  end
+
+	def gameplay
+    change_turn until win?(@player_1.symbol) || win?(@player_2.symbol) || @board.full?
+    winner
+  end
+
+  def winner
+    puts ' Player 1 wins' if win?(@player_1.symbol)
+    puts ' Player 2 wins' if win?(@player_2.symbol)
+  end
+
+  def change_turn
```

Comment:
There is repetition here. We can extract the logic inside the `if` and `else` block into its own method and just inject the appropriate dependencies when you call that method here.

Also there are some lines that should be part of Player's responsibility which you should just be invoking in here.

For instance:

`@player_1.positions.push(index)` should be part of a method inside the Player class instead of working all that logic in Game.rb.